### PR TITLE
Don't allow placing cards in setup before the game is started

### LINKED ISSUE #826: Cards can be played before keep/mulligan decision is made
Because of bad habbits I occasionally mulligan while having marshalled half way. 
That means that the cards i have left on board will not get mulliganed/shuffled/drawn. 

Its not much of an issue, but sometimes it bugs me :)

### PR DESCRIPTION
**Pull request by cryogen**
Also, clear the cards in play if a mulligan is taken so all cards are shuffled

Fixes #826

### CODE REVIEW

File: server/game/setupcardaction.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ class SetupCardAction extends BaseAbility {
 
     meetsRequirements(context) {
         return (
+            context.game.readyToStart &&
```
Comment (ystros) [inline server/game/setupcardaction.js]: This variable doesn't seem to be set outside of tests??? There's a Player.readyToStart but not at the game level.

File: server/game/player.js [hunk 2]
```diff
@@ -310,7 +310,14 @@ class Player extends Spectator {
             card.moveTo('draw deck');
             this.drawDeck.push(card);
         });
+        this.cardsInPlay.each(card => {
```
Comment (ystros) [inline server/game/player.js]: Probably should move things out of discard and dead pile as well, because someone somewhere is going to do that.

Comment (zacchiro) [inline server/game/player.js]: While we're at it, I'd appreciate if we can rename/generalize this method into something like resetDrawDeck(), if not even resetDecks() (plural) makingit also take care of the plot deck. I plan to use this to implement a /reset chat command, which we've discussed before, and it'd be nice to have most of its logic part of a method also used for other purposes, such as game initialization at startup.

File: server/game/setupcardaction.js (hunk 1 above)
Comment (cryogen) [inline server/game/setupcardaction.js]: Github needs more reaction emojis, my screwups are not accurately reactable in the current list

File: server/game/player.js [hunk 3]
```diff
@@ -305,12 +305,36 @@ class Player extends Spectator {
...
+        this.resetCardPile(this.discardPile);
+        this.discardPile = _([]);
+
+        this.resetCardPile(this.deadPile);
+        this.deadPile = _([]);
+
+        _.each(this.additionalPiles, pile => {
+            this.resetCardPile(_(pile.cards));
+        });
+        this.additionalPiles = {};
```
Comment (ystros) [inline server/game/player.js]: I'm pretty sure this will kill the Rains schemes plot.

Comment (cryogen) [reply server/game/player.js]: In my testing it doesn't appear to

**Pull request merged**